spec/jobs/suitable_for_early_allocation_email_job_spec.rb

```diff
@@ -0,0 +1,114 @@
+require 'rails_helper'
+
+RSpec.describe SuitableForEarlyAllocationEmailJob, :allocation, type: :job do
+  let(:offender) do
+    build(:offender, :determinate, latestLocationId: 'LEI',
+          sentence: build(:sentence_detail,
+                          sentenceStartDate: Time.zone.today - 10.months,
+                          conditionalReleaseDate: release_date,
+                          automaticReleaseDate: release_date,
+                          releaseDate: release_date,
+                          tariffDate: nil))
+  end
+
+  before do
+    create(:case_information, nomis_offender_id: offender.offender_no)
+    create(:allocation, nomis_offender_id: offender.offender_no)
+    allow(OffenderService).to receive(:get_offender).and_return(offender)
+  end
+
+  context 'when form created within the referral window (within 18 months to release)' do
+    let(:release_date) { Time.zone.today + 8.months }
+
+    it 'does not send email' do
+      early_alloc_form = create(:early_allocation, :discretionary,
+                                created_within_referral_window: offender.within_early_allocation_window?,
+                                nomis_offender_id:  offender.offender_no)
+
+      expect(early_alloc_form.created_within_referral_window).to eq(true)
+      expect_any_instance_of(EarlyAllocationMailer).not_to receive(:review_early_allocation)
+      described_class.perform_now
+    end
+  end
+
+  context 'when form created outside of the referral window (more than 18 months to release)' do
+    context 'when form outcome is ineligible' do
+      let(:release_date) { Time.zone.today + 28.months }
+
+      it 'does not send email' do
+        create(:early_allocation, :ineligible, created_within_referral_window: false, nomis_offender_id: offender.offender_no)
+
+        expect_any_instance_of(EarlyAllocationMailer).not_to receive(:review_early_allocation)
+        described_class.perform_now
+      end
+    end
+
+    context 'when offender is not allocated to a POM' do
+      let(:release_date) { Time.zone.today + 17.months }
+
+      before do
```

Comment:
It's more idiomatic to not create data rather than destroy things like this - the context above probably should say 'with an allocation'

Reply:
Done